moveit_experimental/moveit_jog_arm/src/jog_calcs.cpp

```diff
@@ -386,12 +391,9 @@ bool JogCalcs::convertDeltasToOutgoingCmd()
   if (!enforceSRDFJointBounds(outgoing_command_))
   {
     suddenHalt(outgoing_command_);
-    has_warning_ = true;
+    status_ = kJointBound;
```

Comment:
Wouldn't it be way better if we only publish when there's an issue (relatively rare occurrence) and just replace these status variable updates with publish commands? This would then read `publishStatus(WarningCodeEnum::kJointBound)`. My problem with setting the `status_` variable here and there is that eventually it is not updated correctly and you either miss a warning(s) or get wrong (outdated) warning.

Reply:
I actually prefer to publish once per iteration. If you only publish an error, the subscriber won't be sure how long ago the error occurred and if it's resolved.

I can delete `status_` and just call `publishStatus()` immediately. I think that makes sense.